pkg/kv/kvserver/allocator_scorer.go

```diff
@@ -1856,6 +1997,73 @@ func convergesOnMean(oldVal, newVal, mean float64) bool {
 	return math.Abs(newVal-mean) < math.Abs(oldVal-mean)
 }
 
+type diskHealthScorerOptions struct {
+	enforcementLevel    storeHealthEnforcement
+	l0SublevelThreshold int64
+}
+
+func (o *diskHealthScorerOptions) getDiskHealthScorerOptions() *diskHealthScorerOptions {
+	return o
+}
+
+// readAmpCheck returns true if the store read amplification does not exceed
+// the cluster threshold and mean, or the enforcement level does not include
+// excluding candidates from being allocation targets.
+func readAmpCheck(
+	ctx context.Context, options *diskHealthScorerOptions, store roachpb.StoreDescriptor, avg float64,
+) bool {
+	if options.enforcementLevel == storeHealthNoAction ||
+		store.Capacity.L0Sublevels < options.l0SublevelThreshold {
+		return true
+	}
+
+	// Still log an event when the L0 sub-levels exceeds the threshold, however
+	// does not exceed the cluster average. This is enabled to avoid confusion
+	// where candidate stores are still targets, despite exeeding the
+	// threshold.
+	if float64(store.Capacity.L0Sublevels) < avg*l0SubLevelWaterMark {
+		log.Eventf(ctx, "s%d, allocate check l0 sublevels %d exceeds threshold %d, but below average: %f, action enabled %d",
+			store.StoreID, store.Capacity.L0Sublevels,
+			options.l0SublevelThreshold, avg, options.enforcementLevel)
```

Comment:
should probably change that to mention the watermark

Reply:
Good idea. Updated.